lib/src/components/board/board.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:viam_sdk/src/resource/base.dart';
+
+import '../../gen/common/v1/common.pb.dart' as common;
+import '../../robot/client.dart';
+
+class BoardStatus {
+  final Map<String, int> analogs;
+  final Map<String, int> digitalInterrupts;
+
+  BoardStatus(this.analogs, this.digitalInterrupts);
+
+  factory BoardStatus.fromProto(common.BoardStatus pbBoardStatus) {
+    BoardStatus boardStatus = BoardStatus(Map<String, int>(), Map<String, int>());
+    pbBoardStatus.analogs.forEach((key, value) => boardStatus.analogs[key] = value.value);
+    pbBoardStatus.digitalInterrupts.forEach((key, value) => boardStatus.digitalInterrupts[key] = value.value.toInt());
+    return boardStatus;
+  }
+}
+
+enum ViamPowerMode {
```

Comment:
Naming this just `PowerMode` caused some problems and i'd need to add a prefix which would be used in a lot of places in this file if I wanted to keep the naming consistent. I chose to call this `ViamPowerMode` instead, because the prefix felt heavy handed. Open to going the other route if this isn't a naming scheme we want to follow.

Reply:
What's the "naming problem"? Is there another variable called PowerMode?

Reply:
this is the error btw https://dart.dev/tools/diagnostic-messages?utm_source=dartdev&utm_medium=redir&utm_id=diagcode&utm_content=ambiguous_import#ambiguous_import

```
The name 'PowerMode' is defined in the libraries 'package:viam_sdk/src/components/board/board.dart' and 'package:viam_sdk/src/gen/component/board/v1/board.pbenum.dart'.
Try using 'as prefix' for one of the import directives, or hiding the name from all but one of the imports.
```

Reply:
I looked into adding the prefixes again, thinking i may be able to only add it to one, but it still insists I add a prefix to both, which would mean adding a prefix to everything coming into client.dart from board.dart, which is pretty much everything. I think I am back to where I was before of feeling that is heavy handed and may want to punt this decision a bit to think about where to move the enum to avoid this or some other solution 🤔

Reply:
Sounds good to me. That's some helpful error messaging from `dart`.

Reply:
Why can't we just use the generated `PowerMode`? Why do we need to create our own?

Reply:
I was trying to remove any direct references to pb/generated code. I guess it being an enum opens the door to passing in the enum value or something. I could look into that.

Reply:
I don't think we need to remove direct refs to generated code, as long as the generated code fits within the paradigm of the native language. So I think using the generated enum would be ok